### PR TITLE
Simplify `package.json` parsing

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by sschuberth** [comment]
@fviernau, just some small follow-up changes that should not interfere with your upcoming changes.

File: plugins/package-managers/node/src/main/kotlin/PackageJson.kt [hunk 1]
```diff
@@ -143,15 +143,14 @@ private object AuthorListSerializer : JsonTransformingSerializer<List<Author>>(s
...
-        takeUnless { this is JsonPrimitive } ?: JsonObject(mapOf("name" to jsonPrimitive))
 }
 
 private object RepositorySerializer : JsonTransformingSerializer<Repository>(serializer<Repository>()) {
-    override fun transformDeserialize(element: JsonElement): JsonElement =
-        (element as? JsonObject) ?: JsonObject(mapOf("url" to element.jsonPrimitive))
+    override fun transformDeserialize(element: JsonElement): JsonElement = element.wrapPrimitiveInObject("url")
 }
+
+private fun JsonElement.wrapPrimitiveInObject(key: String) =
```
Comment (fviernau) [inline plugins/package-managers/node/src/main/kotlin/PackageJson.kt:156]: The function is not re-used in this PR. Why is the extraction needed?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/package-managers/node/src/main/kotlin/PackageJson.kt (hunk 1 above)
Comment (sschuberth) [reply plugins/package-managers/node/src/main/kotlin/PackageJson.kt:156]: The function *is* reused / shared across `AuthorListSerializer` and `RepositorySerializer` (and potential future serializers, though I have no plans for any).

**Review by fviernau — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the unrelated `ClearlyDefinedServiceFunTest` timeout.

**Pull request merged**